--- a/gameMenu/MenuPanelGui.java
+++ b/gameMenu/MenuPanelGui.java
@@ -135,7 +135,14 @@
 
             }
         }));
-        this.buttons.add(this.addLine(7, 13, 2, GameText.getText(6), new ClickListener() {
+        // Evolve start
+        this.buttons.add(this.addLine(7, 13, 2, "Mods", event -> {
+            if (event.isLeftClick()) {
+                this.gameMenu.setNewSecondaryScreen(new wtf.choco.evolve.gui.mods.ModPanelGui(gameMenu));
+            }
+        }));
+        this.buttons.add(this.addLine(8, 15, 2, GameText.getText(6), new ClickListener() {
+        // Evolve end
             public void eventOccurred(GuiClickEvent guiclickevent) {
                 if (guiclickevent.isLeftClick()) {
                     SoundMaestro.getMusicPlayer().fadeOutAndStopPlayer();
