--- a/basics/DisplayManager.java
+++ b/basics/DisplayManager.java
@@ -62,7 +62,7 @@
             Display.setIcon(IconLoader.load(DisplayManager.ICON16_FILE, DisplayManager.ICON32_FILE, DisplayManager.ICON128_FILE));
             Display.setVSyncEnabled(GraphicsOptions.VSYNC);
             Display.setInitialBackground(1.0F, 1.0F, 1.0F);
-            Display.setTitle(GameText.getText(1));
+            Display.setTitle(GameText.getText(1) + " (Modded)"); // Evolve
             Display.create(new PixelFormat(), contextattribs);
             DisplayManager.WIDTH = Display.getWidth();
             DisplayManager.HEIGHT = Display.getHeight();
